docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,340 @@
+could be done at higher-level, creating Analysis instances and running them for each time bin.
+This exercise of pro / con is left to the PIG on Light Curves. We could expect that restructuring
+all of Gammapy to be tools based with good tool chains and config handling will be eventually
+achieved after a certain time if we define a solid high-level interface API.
+
+We could explore the use of `papermill`_ to run workflows defined in a notebook using the values of
+the paramter-value configuration file defined for the high-level interface API. The notebooks could
+be provided as skeleton-templates for specific use cases or built by the user. This option 
+would provide a rich-text formatted report of the analysis process execution.
+
+One extra dimension that arises from the development of this high-level interface API is the possibility
+to capture data provenance as structured logs of tasks executions in a session or in an automated workflow. 
+Capturing provenance in this way is only useful if we also provide the means for it to be easily queried 
+and inspected in multiple ways, allowing also forensic studies of the research analysis process as well 
+as improving reproducibility and reuse by the community. This work will be the scope of another PIG.
+
+
+Alternatives
+============
+
+A different approach to this high-level interface API is that of command line tools executed from the 
+terminal, what  `Fermitools`_ and `ctools`_ do, where each tool is simple/atomic enough to allow users
+to inspect the output results before taking a decission on how to run and set the parameter values for
+the next tool. A similar approach could be done with the Gammapy high-level interface API, but inside
+a notebook or IPython session.
+
+Another config-file based solution is what is implemented in `Enrico`_. It performs basic orchestrated
+analysis workflows using a set of input parameters that the user provides via a configuration file. 
+The user may be guided in the declaration of values for the config file using an assistant command line
+tool for config-file building, which asks for parameter values providing also defaults. This is done in 
+`Enrico`_ with  ``enrico_config`` and ``enrico_xml``, where each workflow is set-up and then run with its 
+own command line tool. In our case, we define the workflow steps in a simple Python script and declare 
+parameter-value pairs in a configuration file. The Python script is then executed to run the workflow. 
+
+Also Python scripts and/or notebook files could be generated with an assistant command line tool. Then, 
+the usercould edit and tweak the config files, scripts or notebooks. There isn't much precedence for this
+workflow in science, but a lot of devops and programming tools work like that, it is a standard technique. 
+One random example of such a tool is the `Angular CLI`_, or `cookiecutter`_. 
+
+Task list
```

Comment:
Suggest to mention somewhere that we'll remove `gammapy.image.bin`, and also the very long text I wrote in the `gammapy.scripts` docs. We would keep click and `gammapy info` though.

Maybe rename `gammapy.scripts` to `gammapy.analysis`?